### DISCUSSION TITLE
Add private linter to golangci-lint

### DISCUSSION THREAD

**Discussion by behzadsh** [golangci/golangci-lint#3259] [Q&A] [answered]
Hi everyone.

I created a custom linter using `go/analysis` and followed the [documentation](https://golangci-lint.run/contributing/new-linters/#how-to-add-a-private-linter-to-golangci-lint). I built my plugin and build golangci-lint separately, added the custom linter config to `.golangci.yml` of my project. But when I run the linter I get the following error

```
ERRO Unable to load custom analyzer mylinter:mylinter.so, plugin.Open("/path/to/mylinter"): plugin was built with a different version of package golang.org/x/sys/execabs
```

There's also another discussion with no answer.
https://github.com/golangci/golangci-lint/discussions/2529

**Comment by ldez** [comment, marked as answer]
Hi,

plugins and golangci-lint must be compiled with the exact same stack, it's related to how Go plugins work.

It's a constraint that is not possible to overcome because it's a Go constraint.

**Reply by behzadsh** [reply]
what do you mean by "exact same stack"? I built both of them on the same machine.
Can you explain a little more?

**Reply by ldez** [reply]
From my memory:
- exact same Go version
- same OS/arch
- same version of shared dependencies

In your case, you are using a different version of `golang.org/x/sys` than golangci-lint.

**Reply by behzadsh** [reply]
Ok, thanks. based on what you said I think the problem is the third one.
I'll check it. thanks again.
